sharding/indexHashedNodesCoordinator.go

```diff
@@ -803,22 +426,16 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartAction(hdr data.HeaderHandler
 	newEpoch := hdr.GetEpoch()
 	epochToRemove := int32(newEpoch) - nodesCoordinatorStoredEpochs
 	needToRemove := epochToRemove >= 0
-	ihgs.currentEpoch = newEpoch
+	ihgs.SetCurrentEpoch(newEpoch)
 
 	err := ihgs.saveState(ihgs.savedStateKey)
 	if err != nil {
 		log.Error("saving nodes coordinator config failed", "error", err.Error())
 	}
 
-	ihgs.mutNodesConfig.Lock()
 	if needToRemove {
-		for epoch := range ihgs.nodesConfig {
-			if epoch <= uint32(epochToRemove) {
-				delete(ihgs.nodesConfig, epoch)
-			}
-		}
+		ihgs.RemoveNodesConfigEpochs(epochToRemove)
```

Comment:
it is in fact the same since RemoveNodesConfigEpochs uses the mutex
needToRemove check does not need to be protected, so the behavior should be similar